### ISSUE TITLE
Upgrade to new Blazor version

### ISSUE THREAD

**Issue by stsrki** [stsrki/Blazorise#68]
https://devblogs.microsoft.com/aspnet/blazor-now-in-official-preview/

https://devblogs.microsoft.com/aspnet/asp-net-core-updates-in-net-core-3-0-preview-4/

**Comment by VR-Architect** [comment]
If you need another test for core 3 preview 4, let me know.

**Comment by stsrki** [comment]
All of the projects are converted to preview 4. There are only two issues left now.

1. Razor Components for Server-side Blazor demo application are giving me timeout when I try to navigate to another page in application.
2. One of unit tests are failing.

If anyone is willing to help me with first problem I would be really grateful. So far the changes are still on my local repo since there are over 500 files changed for the new preview 4. I can push it if anyone is willing to have a look.

**Comment by neven10** [comment]
Timeout as in it shows "Reconnecting to server"?
Or they don't render at all?

**Comment by stsrki** [comment]
Yes, "Reconnecting to server". I think the problem could be on Dropdown control due to the JSInterop calling async from sync code.

**Comment by neven10** [comment]
Push changes if you have time,  i'll try to test it today.

**Comment by stsrki** [comment]
Pushed, the code is on branch dev070

**Comment by neven10** [comment]
Tried Dropdown and few other components and navigating works fine. Can you show me or point me to the code sample where it breaks on navigation, maybe i'm not on the right track.

**Comment by stsrki** [comment]
The problem can be seen on project Blazorise.Demo.Bootstrap.RC. Just try to click on pages on the sidebar and you should see the problem.

**Comment by neven10** [comment]
Yep, confirmed on Blazorise.Demo.Bootstrap.RC .

Here's what i did. 
I created new templated project and referenced what's necessary.
I copied pages for Buttons, Cards and Forms from   Blazorise.Demo.Bootstrap.RC and added links to default NavMenu.
Smashed between pages and no timeouts. So that works fine.

I'll have to poke around more and see what's wrong on Demo projects.

**Comment by neven10** [comment]
Dropdown :
```
Exception thrown: 'System.NullReferenceException' in Blazorise.dll
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.0.0-preview4-27615-11\System.Diagnostics.StackTrace.dll'. 
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.0.0-preview4-27615-11\System.Reflection.Metadata.dll'. 
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.0.0-preview4-27615-11\System.Collections.Immutable.dll'. 
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.0.0-preview4-27615-11\System.IO.MemoryMappedFiles.dll'. 
Microsoft.AspNetCore.Components.Browser.Rendering.RemoteRenderer: Warning: Unhandled exception rendering component: Object reference not set to an instance of an object.

System.NullReferenceException: Object reference not set to an instance of an object.
   at Blazorise.Base.BaseDropdown.set_IsOpen(Boolean value) in C:\Users\Neven\Desktop\GithubProj\Blazorise\Source\Blazorise\Base\BaseDropdown.cs:line 155
   at Blazorise.Base.BaseDropdown.Toggle() in C:\Users\Neven\Desktop\GithubProj\Blazorise\Source\Blazorise\Base\BaseDropdown.cs:line 86
   at Blazorise.Base.BaseDropdownToggle.ClickHandler() in C:\Users\Neven\Desktop\GithubProj\Blazorise\Source\Blazorise\Base\BaseDropdownToggle.cs:line 63
   at Microsoft.AspNetCore.Components.EventCallbackWorkItem.InvokeAsync[T](MulticastDelegate delegate, T arg)
   at Microsoft.AspNetCore.Components.ComponentBase.Microsoft.AspNetCore.Components.IHandleEvent.HandleEventAsync(EventCallbackWorkItem callback, Object arg)
   at Microsoft.AspNetCore.Components.EventCallback.InvokeAsync(Object arg)
   at Blazorise.Base.BaseButton.ClickHandler() in C:\Users\Neven\Desktop\GithubProj\Blazorise\Source\Blazorise\Base\BaseButton.cs:line 67
   at Microsoft.AspNetCore.Components.EventCallbackWorkItem.InvokeAsync[T](MulticastDelegate delegate, T arg)
   at Microsoft.AspNetCore.Components.ComponentBase.Microsoft.AspNetCore.Components.IHandleEvent.HandleEventAsync(EventCallbackWorkItem callback, Object arg)
   at Microsoft.AspNetCore.Components.Rendering.Renderer.DispatchEventAsync(Int32 eventHandlerId, UIEventArgs eventArgs)
Microsoft.AspNetCore.Components.Server.ComponentHub: Warning: Unhandled Server-Side exception

System.NullReferenceException: Object reference not set to an instance of an object.
   at Blazorise.Base.BaseDropdown.set_IsOpen(Boolean value) in C:\Users\Neven\Desktop\GithubProj\Blazorise\Source\Blazorise\Base\BaseDropdown.cs:line 155
   at Blazorise.Base.BaseDropdown.Toggle() in C:\Users\Neven\Desktop\GithubProj\Blazorise\Source\Blazorise\Base\BaseDropdown.cs:line 86
   at Blazorise.Base.BaseDropdownToggle.ClickHandler() in C:\Users\Neven\Desktop\GithubProj\Blazorise\Source\Blazorise\Base\BaseDropdownToggle.cs:line 63
   at Microsoft.AspNetCore.Components.EventCallbackWorkItem.InvokeAsync[T](MulticastDelegate delegate, T arg)
   at Microsoft.AspNetCore.Components.ComponentBase.Microsoft.AspNetCore.Components.IHandleEvent.HandleEventAsync(EventCallbackWorkItem callback, Object arg)
   at Microsoft.AspNetCore.Components.EventCallback.InvokeAsync(Object arg)
   at Blazorise.Base.BaseButton.ClickHandler() in C:\Users\Neven\Desktop\GithubProj\Blazorise\Source\Blazorise\Base\BaseButton.cs:line 67
   at Microsoft.AspNetCore.Components.EventCallbackWorkItem.InvokeAsync[T](MulticastDelegate delegate, T arg)
   at Microsoft.AspNetCore.Components.ComponentBase.Microsoft.AspNetCore.Components.IHandleEvent.HandleEventAsync(EventCallbackWorkItem callback, Object arg)
   at Microsoft.AspNetCore.Components.Rendering.Renderer.DispatchEventAsync(Int32 eventHandlerId, UIEventArgs eventArgs)
Exception thrown: 'System.NullReferenceException' in Microsoft.AspNetCore.Components.dll
Exception thrown: 'System.NullReferenceException' in Microsoft.AspNetCore.Components.dll
Exception thrown: 'System.AggregateException' in System.Private.CoreLib.dll
Exception thrown: 'System.AggregateException' in System.Private.CoreLib.dll
```

Again this only happens on Demo, when i use same thing in fresh project , there's no problem.

**Comment by stsrki** [comment]
Excellent work! I will now try to find why is there NullReferenceException happening. Must be something I forgot to null-check. tnx

**Comment by stsrki** [comment]
@neven10  Can you tell how did you get the stack trace?

**Comment by neven10** [comment]
It will  appear on the output window of Visual Studio. Make sure in debug you have CLR Exceptions checked. 
It won't appear every time. For example, try selecting dropdown in FormsPage , after page is finished rendering (after it reconnects) or before that and then navigate to something else.

As for catching it and throwing manually to get stack trace  , i'm not really sure. OnInit method?

**Comment by stsrki** [comment]
It's really hard to find whats causing the error. Even when I remove all of the drop-downs it still happens randomly, although much less frequently. I'm thinking of just ignoring it for v0.7.0 and fix it later in v0.7.1

**Comment by stsrki** [comment]
It's really hard to find whats causing the error. Even when I remove all of the drop-downs it still happens randomly, although much less frequently. I'm thinking of just ignoring it for v0.7.0 and fix it later in v0.7.1

**Comment by stsrki** [comment]
@neven10

The issue was with **SignalR** messages because they are too small by default, around 32KB. In case anyone ever stumble upon this same problem here is the solution. 

This should be placed at Startup.

```
services.AddServerSideBlazor().AddSignalR().AddHubOptions<ComponentHub>( o =>
{
    o.MaximumReceiveMessageSize = 1024 * 1024 * 100;
} );
```

```
app.UseSignalR( route => route.MapHub<ComponentHub>( ComponentHub.DefaultPath, o =>
{
    o.ApplicationMaxBufferSize = 1024 * 1024 * 100; // larger size
    o.TransportMaxBufferSize = 1024 * 1024 * 100; // larger size
} ) );
```
